coldfront/plugins/qumulo/tests/utils/test_eib_billing.py

```diff
@@ -0,0 +1,497 @@
+        self.assertTrue(re.search("RIS-[A-Za-z]+-storage2-active-billing.csv", filename))
+        self.assertTrue(os.path.exists(filename))
+        os.system(f"ls -l {filename}")
+
+        with open(filename) as csvreport:
+            data = list(csv.reader(csvreport))
+
+        # Confirm the billing amount for 5 unit of Consumption cost model is $65 
+        # hardcoded
+        billing_amount = float(data[len(data)-1][19])
+        self.assertEqual(billing_amount - 65.0, 0)
+
+        os.remove(filename)
+
+
+    # @skip("skip for now")
+    @patch("coldfront.plugins.qumulo.tasks.QumuloAPI")
+    def test_create_allocations_ingest_usages_generate_billing_report(self, qumulo_api_mock: MagicMock) -> None:
+        qumulo_api = MagicMock()
+        qumulo_api.get_all_quotas_with_usage.return_value = mock_get_quotas_2()
+        qumulo_api_mock.return_value = qumulo_api
+
+        allocations = [
+            set_allocation_form_data_with_5TB_consumption,
+            set_allocation_form_data_with_15TB_consumption,
+            set_allocation_form_data_with_100TB_consumption,
+            set_allocation_form_data_with_5TB_subscription,
+            set_allocation_form_data_with_100TB_subscription,
+            set_allocation_form_data_with_500TB_subscription,
+            set_allocation_form_data_with_500TB_subscription_500tb,
+            set_allocation_form_data_with_500TB_condo,
+        ]
+
+        for allocation in allocations:
+            create_allocation(
+                project=self.project,
+                user=self.user,
+                form_data=allocation()
+            )
+
+        # Update the status of the created allocations from Pending to Active
+        for allocation in Allocation.objects.all():
+            if (allocation.resources.first().name == "Storage2"):
+                allocation.status=AllocationStatusChoice.objects.get(name="Active")
+                allocation.save()
+
+        storage2_allocations = []
+        for allocation in Allocation.objects.all():
+            if (allocation.resources.first().name == "Storage2" and
+                allocation.status.name == "Active"):
+                storage2_allocations.append(allocation) 
+
+        self.assertEqual(len(storage2_allocations), len(allocations))
+
+        ingest_quotas_with_daily_usage()
+        generate_monthly_billing_report(datetime.now(timezone.utc).strftime('%Y-%m-%d'))
+
+        filename = get_filename()
+        with open(filename) as csvreport:
+            data = list(csv.reader(csvreport))
+
+        header = get_report_header()
+        num_lines_header = len(header.splitlines())
+        print("Numer of lines of the report header: %s" % num_lines_header)
+        self.assertEqual(num_lines_header, 5)
+
+        row = 1
+        for idx in range(num_lines_header, len(data)):
+            # print("Row%s: %s" % (idx, data[idx]))
+            row_num = data[idx][1]
+            self.assertEqual(str(row), row_num)
```

Comment:
could replace `row` with `index+1` and then not need to worry about incrementing `row`

Reply:
`index + 1` is the line number of the report including 5 lines of headers. `row` is the item number of the billing entries starting from 1. `row` = `index + 1` - 5. I think using a separate counter would make more sense.